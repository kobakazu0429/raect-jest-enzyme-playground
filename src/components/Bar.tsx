import React, { useState, useCallback, ChangeEvent } from "react";

export interface IProps {
  name: string;
  age?: number;
}

export const Bar = ({ name, age: _age = 0 }: IProps) => {
  const [age, setAge] = useState(_age);
  const handleChange = useCallback(
    (e: ChangeEvent<HTMLInputElement>) => {
      setAge(Number(e.target.value));
    },
    [age]
  );

  return (
    <div>
      <p id="name">name: {name}</p>
      <p id="age">age: {age}</p>
      <input type="number" value={age} onChange={handleChange} />
    </div>
  );
};
