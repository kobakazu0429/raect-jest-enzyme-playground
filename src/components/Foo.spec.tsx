import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";

import { Foo } from "@/components/Foo";

describe("<Foo />", () => {
  it("pタグが1つある", () => {
    const wrapper = shallow(<Foo />);
    expect(wrapper.find("p").length).toBe(1);
  });

  it("should match the snapshot", () => {
    const tree = renderer.create(<Foo />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
