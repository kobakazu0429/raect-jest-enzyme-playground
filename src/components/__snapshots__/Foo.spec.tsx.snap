// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Foo /> should match the snapshot 1`] = `
<p>
  Foo
</p>
`;
